velero/backup/common-service/label-common-service.sh

```diff
@@ -0,0 +1,177 @@
+    # Label the Private CatalogSources in provided namespaces
+    if [ $ENABLE_PRIVATE_CATALOG -eq 1 ]; then
+        CS_SOURCE_NS=$OPERATOR_NS
+        CM_SOURCE_NS=$CERT_MANAGER_NAMESPACE
+        LIS_SOURCE_NS=$LICENSING_NAMESPACE
+        LSR_SOURCE_NS=$LSR_NAMESPACE
+        EDB_SOURCE_NS=$OPERATOR_NS
+
+        ${OC} label catalogsource ibm-operator-catalog foundationservices.cloudpak.ibm.com=catalog -n $DEFAULT_SOURCE_NS --overwrite=true 2>/dev/null 
+        ${OC} label catalogsource $CS_SOURCE foundationservices.cloudpak.ibm.com=catalog -n $CS_SOURCE_NS --overwrite=true 2>/dev/null 
+        ${OC} label catalogsource $CM_SOURCE foundationservices.cloudpak.ibm.com=catalog -n $CM_SOURCE_NS --overwrite=true 2>/dev/null 
+        ${OC} label catalogsource $LIS_SOURCE foundationservices.cloudpak.ibm.com=catalog -n $LIS_SOURCE_NS --overwrite=true 2>/dev/null 
+        ${OC} label catalogsource $EDB_SOURCE foundationservices.cloudpak.ibm.com=catalog -n $EDB_SOURCE_NS --overwrite=true 2>/dev/null
+    fi
+
+    # Label the CatalogSource with .spec.publisher having "IBM" in openshift-marketplace namespace
+    local ibm_catalog_sources=""
+    while IFS=' ' read -r -a sources; do
+        for source in "${sources[@]}"; do
+            if ${OC} get catalogsource "$source" -n "$DEFAULT_SOURCE_NS" -o json | grep -q '"publisher": *"IBM"*'; then
+                ibm_catalog_sources+=" $source"
+            fi
+        done
+    done <<< "$(${OC} get catalogsource -n "$DEFAULT_SOURCE_NS" -o jsonpath='{.items[*].metadata.name}')"
+
+    # Remove leading/trailing whitespace
+    ibm_catalog_sources=$(echo "${ibm_catalog_sources}" | tr -s ' ' | sed 's/^ *//g' | sed 's/ *$//g')
+    for source in $ibm_catalog_sources; do
+        ${OC} label catalogsource "$source" foundationservices.cloudpak.ibm.com=catalog -n $DEFAULT_SOURCE_NS --overwrite=true 2>/dev/null
+    done
+}
+
+function label_ns_and_related() {
+
+    title "Start to label the namespaces, operatorgroups and secrets... "
+    namespaces=$(${OC} get configmap namespace-scope -n $OPERATOR_NS -oyaml | awk '/^data:/ {flag=1; next} /^  namespaces:/ {print $2; next} flag && /^  [^ ]+: / {flag=0}')
+    # add cert-manager namespace and licensing namespace and lsr namespace into the list with comma separated
+    namespaces+=",$CONTROL_NS,$CERT_MANAGER_NAMESPACE,$LICENSING_NAMESPACE,$LSR_NAMESPACE"
+    namespaces=$(echo "$namespaces" | tr ',' '\n')
+
+    while IFS= read -r namespace; do
+        # Label the namespace
+        ${OC} label namespace "$namespace" foundationservices.cloudpak.ibm.com=namespace --overwrite=true 2>/dev/null
+        
+        # Label the OperatorGroup
+        operator_group=$(${OC} get operatorgroup -n "$namespace" -o jsonpath='{.items[*].metadata.name}')
+        ${OC} label operatorgroup "$operator_group" foundationservices.cloudpak.ibm.com=operatorgroup -n "$namespace" --overwrite=true 2>/dev/null
+        
+        # Label the entitlement key
```

Comment:
We may need to include the catalogsource namespaces in the group of namespaces to iterate through. I typically deploy an entitlement key secret in the openshift-marketplace namespace and I noticed that it was not covered after running the script